### PR TITLE
Update openapi-authoring-automated-guidelines.md

### PR DESCRIPTION
**Pull request by jianyexi**
Changes:
update documentation for R2005
add new rule: MissingTypeObject

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md [hunk 1]
```diff
@@ -1907,7 +1907,7 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 In case of LRO Post operation with return schema, it MAY be ambiguous for the SDK to understand automatically what the return schema is modeling. To avoid any confusion that would lead SDK to incorrectly instantiate the return type, service team needs to explain if the return schema is modeling a result from a "Location" header, or from an "Azure-AsyncOperation" header.
 
-More details on LRO operation could be found [here](https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md)
+More details on LRO operation could be found [here](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation)
```
Comment (raych1) [inline documentation/openapi-authoring-automated-guidelines.md:1910]: The original link is the document of how SDK handles LRO. Should it be moved somewhere?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 2]
```diff
@@ -3459,4 +3459,61 @@ GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{
...
+
+### <a name="r4037"></a>R4037 MissingTypeObject
+
+**Category** : SDK Error
+
+**Applies to** : ARM and Data plane OpenAPI(swagger) specs
+
+**Output Message** : The schema '{json path}' is considered an object but without a 'type:object', please add the missing 'type:object'.
+
+**Description**: This rule ensure that any schema whose type is 'object' must defines its type explicity, as a schema without type indicates a free-form schema which means it can be any type. For this rule, a schema with keyword 'properties' or 'additionalProperties' is considered an object schema.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3473]: Description： The rule is to ensure that any 'object' type schema must be defined with "type":"object" explicitly. Without "type" definition indicates the schema can be any type.  Especially, a schema with keyword 'properties' or ‘addtionalProperties’ requires definition of "type":"object".

File: documentation/openapi-authoring-automated-guidelines.md [hunk 3]
```diff
@@ -3459,4 +3459,61 @@ GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{
...
+
+**CreatedAt**: May 24, 2021
+
+**LastModifiedAt**: May 24, 2021
+
+**Why this rule is important**: It's used to differentiate free-form object and any type , as the SDK track1 doesn’t make a difference between the two syntax.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3479]: Why this rule is important: It's to differentiate object type from any type in SDK Track 2. While the SDK Track1 generator does not make difference between with and without "type":"object" definition (all treated as "object"), the SDK Track 2 generator requires explicit definition for  object types.

**Review by josefree — APPROVED** [review]

**Pull request merged**